src/rpc_server.rs

```diff
@@ -3064,6 +3111,64 @@ impl RPC for NeptuneRPCServer {
         Ok(())
     }
 
+    // documented in trait. do not add doc-comment.
+    async fn pow_solution(
+        self,
+        _context: tarpc::context::Context,
+        token: rpc_auth::Token,
+        nonce: Digest,
+    ) -> RpcResult<bool> {
+        log_slow_scope!(fn_name!());
+        token.auth(&self.valid_tokens)?;
+
+        let Some(mut proposal) = self
+            .state
+            .lock_guard()
+            .await
+            .block_proposal
+            .map(|x| x.to_owned())
+        else {
+            warn!(
+                "Got claimed PoW solution but no challenge was known. \
+            Did solution come in too late?"
+            );
+            return Ok(false);
+        };
+
+        // A proposal was found. Check if solution works.
+        let (guesser_key_after_image, latest_block_header) = {
+            let state = self.state.lock_guard().await;
+            let guesser_key_after_image = state
+                .wallet_state
+                .wallet_secret
+                .guesser_spending_key(proposal.header().prev_block_digest);
+            let latest_block_header = *state.chain.light_state().header();
+
+            (guesser_key_after_image.after_image(), latest_block_header)
+        };
+
+        proposal.set_header_guesser_digest(guesser_key_after_image);
+        proposal.set_header_nonce(nonce);
+        let threshold = latest_block_header.difficulty.target();
+        let solution_digest = proposal.hash();
+        if solution_digest > threshold {
+            warn!(
+                "Got claimed PoW solution but PoW threshold was not met.\n\
+            Claimed solution: {solution_digest};\nthreshold: {threshold}"
+            );
+            return Ok(false);
+        }
+
+        // No time to waste! Inform main_loop!
+        let solution = Box::new(proposal);
+        let _ = self
```

Comment:
Should a fail to send the message actually be ignored? A closed channel might indicate the application is in a critical state and should be terminated. If this is not case, it might be worth explaining why a send failure can be ignored.